src/code/audio/code_8019AF00.c

```diff
@@ -4463,35 +4744,238 @@ void Audio_ClearSariaBgm2(void) {
+                    } else {
+                        volumeFadeInTimer = gActiveSeqs[SEQ_PLAYER_BGM_SUB].volScales[1] - sBgmEnemyVolume;
+                    }
+
+                    Audio_SetVolumeScale(SEQ_PLAYER_BGM_SUB, 3, sBgmEnemyVolume, volumeFadeInTimer);
+                    Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_SUB) << 24) | ((u32)(10) << 0x10) |
+                                      (u32)(NA_BGM_ENEMY | 0x800));
+
+                    if (seqId >= NA_BGM_TERMINA_FIELD) {
+                        Audio_SetVolumeScale(SEQ_PLAYER_BGM_MAIN, 3, 0x7F - sBgmEnemyVolume, 10);
+                        Audio_SplitBgmChannels(sBgmEnemyVolume);
+                    }
+                } else if ((sPrevSeqMode & 0x7F) == SEQ_MODE_ENEMY) {
+                    // If only sPrevSeqMode = SEQ_MODE_ENEMY (End)
+                    Audio_QueueSeqCmd(0x100000FF | ((u8)(SEQ_PLAYER_BGM_SUB) << 24) | ((10) << 16));
+
+                    if (seqMode == SEQ_MODE_IGNORE) {
+                        volumeFadeOutTimer = 0;
+                    } else {
+                        volumeFadeOutTimer = 10;
+                    }
+
+                    Audio_SetVolumeScale(SEQ_PLAYER_BGM_MAIN, 3, 0x7F, volumeFadeOutTimer);
+                    Audio_SplitBgmChannels(0);
+                }
+
+                sPrevSeqMode = seqMode + 0x80;
+            } else {
+                if (seqMode == SEQ_MODE_ENEMY) {
+                    // If both seqMode = sPrevSeqMode = SEQ_MODE_ENEMY
+                    if ((Audio_GetActiveSequence(SEQ_PLAYER_BGM_SUB) == NA_BGM_DISABLED) &&
+                        (seqId != NA_BGM_DISABLED) && (sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_ENEMY)) {
+                        Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_SUB) << 24) | ((u32)(10) << 0x10) |
+                                          (u32)(NA_BGM_ENEMY | 0x800));
+                        sPrevSeqMode = seqMode + 0x80;
+                    }
+                }
+            }
+        } else {
+            // Remnant of OoT's Hyrule Field Sequence
+            if (seqMode == SEQ_MODE_DEFAULT) {
+                if (sPrevSeqMode == SEQ_MODE_STILL) {
+                    sNumFramesMoving = 0;
+                }
+                sNumFramesStill = 0;
+                sNumFramesMoving++;
+            } else {
+                sNumFramesStill++;
+            }
+
+            if ((seqMode == SEQ_MODE_STILL) && (sNumFramesStill < 30) && (sNumFramesMoving > 20)) {
+                seqMode = SEQ_MODE_DEFAULT;
+                if (!seqMode && !seqMode) {}
+            }
+
+            sPrevSeqMode = seqMode;
+            Audio_QueueSeqCmd(0x70000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(2) << 0x10) | (u32)(seqMode));
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A32CC.s")
+void Audio_UpdateEnemyBgmVolume(f32 dist) {
```

Comment:
Is this responsible for everything to do with enemy proximity bgm?

Reply:
The code for enemy proximilty bgm is scattered across z_actor/z_player and here iirc. Also, the music is initialized in `Audio_SetSequenceMode` but updating/adjusting the main volume happens here